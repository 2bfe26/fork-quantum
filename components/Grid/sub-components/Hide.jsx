import PropTypes from 'prop-types';
import styled from 'styled-components';
import { hide } from './shared/media';

const Hide = styled.div`
  ${props => props.phone && hide.phone()}
  ${props => props.tablet && hide.tablet()}
  ${props => props.desktop && hide.desktop()}
  ${props => props.large && hide.large()}
  ${props => props.hd && hide.hd()}
`;

Hide.propTypes = {
  phone: PropTypes.bool,
  tablet: PropTypes.bool,
  desktop: PropTypes.bool,
  large: PropTypes.bool,
  hd: PropTypes.bool,
};

export default Hide;
