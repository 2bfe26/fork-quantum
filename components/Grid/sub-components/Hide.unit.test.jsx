import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Hide from './Hide';

describe('<Hide />', () => {
  it("should hide it's children according to screen size", () => {
    expect(
      renderer.create(<Hide xsmall>Content</Hide>).toJSON(),
    ).toMatchSnapshot();
    expect(
      renderer.create(<Hide small>Content</Hide>).toJSON(),
    ).toMatchSnapshot();
    expect(
      renderer.create(<Hide medium>Content</Hide>).toJSON(),
    ).toMatchSnapshot();
    expect(
      renderer.create(<Hide large>Content</Hide>).toJSON(),
    ).toMatchSnapshot();
    expect(
      renderer.create(<Hide xlarge>Content</Hide>).toJSON(),
    ).toMatchSnapshot();
  });
});
