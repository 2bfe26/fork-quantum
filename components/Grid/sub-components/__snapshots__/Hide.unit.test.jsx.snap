// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hide /> should hide it's children according to screen size 1`] = `
@media (max-width:599px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1440px) {

}

@media (max-width:599px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

<div
  class="c0"
>
  Content
</div>
`;

exports[`<Hide /> should hide it's children according to screen size 2`] = `
@media (max-width:599px) {

}

@media (min-width:600px) and (max-width:1023px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1440px) {

}

@media (max-width:599px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

<div
  class="c0"
>
  Content
</div>
`;

exports[`<Hide /> should hide it's children according to screen size 3`] = `
@media (max-width:599px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1440px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1440px) {

}

@media (max-width:599px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

<div
  class="c0"
>
  Content
</div>
`;

exports[`<Hide /> should hide it's children according to screen size 4`] = `
@media (max-width:599px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1440px) {

}

@media (max-width:599px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

<div
  class="c0"
>
  Content
</div>
`;

exports[`<Hide /> should hide it's children according to screen size 5`] = `
@media (max-width:599px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

@media (min-width:600px) and (max-width:1023px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1440px) {
  .c0 {
    display: none !important;
  }
}

@media (max-width:599px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

<div
  class="c0"
>
  Content
</div>
`;

exports[`<Hide /> should hide it's children according to screen size 6`] = `
@media (max-width:599px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1440px) {

}

@media (max-width:599px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1024px) and (max-width:1439px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

<div
  class="c0"
>
  Content
</div>
`;

exports[`<Hide /> should hide it's children according to screen size 7`] = `
@media (max-width:599px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:1440px) {

}

@media (min-width:600px) and (max-width:1023px) {

}

@media (min-width:1440px) {

}

@media (max-width:599px) {

}

@media (min-width:1024px) and (max-width:1439px) {

}

@media (min-width:600px) and (max-width:1023px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1024px) and (max-width:1439px) {
  .c0 {
    display: none !important;
  }
}

@media (min-width:1440px) {
  .c0 {
    display: none !important;
  }
}

<div
  class="c0"
>
  Content
</div>
`;
